rtos/Thread.cpp

```diff
@@ -334,6 +334,43 @@ osStatus Thread::wait(uint32_t millisec) {
     return osDelay(millisec);
 }
 
+osStatus Thread::wait_until(uint64_t millisec) {
+    // CMSIS-RTOS 2.1.0 and 2.1.1 differ in the time type, which we determine
+    // by looking at the return type of osKernelGetTickCount. We assume
+    // our header at least matches the implementation, so we don't try looking
+    // at the run-time version report. (There's no compile-time version report)
+    if (sizeof(osKernelGetTickCount() == sizeof(uint64_t))) {
```

Comment:
Shouldn't this be
```
sizeof(osKernelGetTickCount()) == sizeof(uint64_t)
```

rather than
```
sizeof(osKernelGetTickCount() == sizeof(uint64_t))
```

Reply:
Whoops! Good spot. Totally failed to grasp what you were saying earlier...